src/NumericalAlgorithms/Interpolation/InterpolationTargetLineSegment.cpp

```diff
@@ -0,0 +1,57 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Interpolation/InterpolationTargetLineSegment.hpp"
+
+namespace intrp {
+namespace OptionHolders {
+
+template <size_t VolumeDim>
+LineSegment<VolumeDim>::LineSegment(std::array<double, VolumeDim> begin_in,
+                                    std::array<double, VolumeDim> end_in,
+                                    size_t number_of_points_in) noexcept
+    : begin(std::move(begin_in)),  // NOLINT
+      end(std::move(end_in)),      // NOLINT
+      number_of_points(number_of_points_in) {}
+// above NOLINT for std::move of trivially copyable type.
+
+template <size_t VolumeDim>
+void LineSegment<VolumeDim>::pup(PUP::er& p) noexcept {
+  p | begin;
+  p | end;
+  p | number_of_points;
+}
+
+template <size_t VolumeDim>
+bool operator==(const LineSegment<VolumeDim>& lhs,
+                const LineSegment<VolumeDim>& rhs) noexcept {
+  return lhs.begin == rhs.begin and lhs.end == rhs.end and
+         lhs.number_of_points == rhs.number_of_points;
+}
+
+template <size_t VolumeDim>
+bool operator!=(const LineSegment<VolumeDim>& lhs,
+                const LineSegment<VolumeDim>& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+// Explicit instantiations
+template struct LineSegment<1>;
```

Comment:
Could you use `GENERATE_INSTANTIATIONS` here? For an example, see `Domain/CreateInitialElement.cpp`